cirq/google/engine/engine_program.py

```diff
@@ -184,7 +184,65 @@ def run_batch(
                                     created_job_id,
                                     self.context,
                                     job,
-                                    batch_mode=True)
+                                    result_type=ResultType.Batch)
+
+    def run_calibration(
+            self,
+            job_id: Optional[str] = None,
+            processor_ids: Sequence[str] = (),
+            description: Optional[str] = None,
+            labels: Optional[Dict[str, str]] = None,
+    ) -> engine_job.EngineJob:
+        """Runs a batch of circuits on the QuantumEngine.
+
+        This method should only be used if the Program object was created
+        with a BatchProgram.  The number of parameter sweeps should match
+        the number of circuits within that BatchProgram.
+
+        This method does not block until a result is returned.  However,
+        no results will be available until the entire batch is complete.
+
+        Args:
+            job_id: Optional job id to use. If this is not provided, a random id
+                of the format 'job-################YYMMDD' will be generated,
+                where # is alphanumeric and YYMMDD is the current year, month,
+                and day.
+            processor_ids: The engine processors that should be candidates
+                to run the program. Only one of these will be scheduled for
+                execution.
+            description: An optional description to set on the job.
+            labels: Optional set of labels to set on the job.
+
+        Returns:
+            An EngineJob. If this is iterated over it returns a list of
+            TrialResults. All TrialResults for the first circuit are listed
+            first, then the TrialResults for the second, etc. The TrialResults
+            for a circuit are listed in the order imposed by the associated
+            parameter sweep.
+        """
+        import cirq.google.engine.engine as engine_base
+        if not job_id:
+            job_id = engine_base._make_random_id('job-')
+        if not processor_ids:
+            raise ValueError('No processors specified')
+
+        # Default run context
```

Comment:
Why pass an empty run context? It shouldn't be required for this case if it isn't used.

Reply:
I think we need to pass something in there, and the engine code I believe validates an empty context.

Reply:
Yes, for now I will keep it in, and we can adjust it later once it changes server-side.